readium/shared/src/main/java/org/readium/r2/shared/publication/Manifest.kt

```diff
@@ -64,6 +64,38 @@ data class Manifest(
         }
     }
 
+    /**
+     * Finds the first [Link] with the given HREF in the manifest's links.
+     *
+     * Searches through (in order) [readingOrder], [resources] and [links] recursively following
+     * alternate and children links.
+     *
+     * If there's no match, try again after removing any query parameter and anchor from the
+     * given [href].
+     */
+    fun linkWithHref(href: String): Link? {
```

Comment:
I moved this from `Publication`, as it makes sense to have it in `Manifest` as well.